mobile/src/bundles/common/components/button/button.tsx

```diff
@@ -1,19 +1,116 @@
-import React from 'react';
-import { Pressable } from 'react-native';
+import React, { useMemo } from 'react';
+import { type PressableProps } from 'react-native';
+import Icon from 'react-native-vector-icons/MaterialIcons';
 
-import { Text } from '~/bundles/common/components/components';
+import { Pressable, Text } from '~/bundles/common/components/components';
+import { globalStyles } from '~/bundles/common/styles/styles';
 
+import { ButtonType, TextCategory } from '../../enums/enums';
 import { styles } from './styles';
 
+type StyleRecord = Record<string, unknown> | undefined | null | false;
+
+type StylesProperties = {
+    style: Record<string, unknown>;
+    pressed: Record<string, unknown>;
+    disabled: Record<string, unknown>;
+    label: Record<string, unknown>;
+};
+
+type ButtonName = (typeof ButtonType)[keyof typeof ButtonType];
+
 type Properties = {
     label: string;
-    onPress: () => void;
-};
+    iconName?: string;
+    iconSize?: number;
+    buttonType?: ButtonName;
+} & PressableProps;
+
+const iconDefaultSize = 32;
+
+const Button: React.FC<Properties> = ({
+    label,
+    style: pressAbleStyle,
+    iconName,
+    iconSize = iconDefaultSize,
+    buttonType = ButtonType.FILLED,
+    disabled = false,
+    ...props
+}) => {
+    const componentStyles: Record<ButtonName, StylesProperties> =
+        useMemo(() => {
+            return {
+                [ButtonType.FILLED]: {
+                    style: styles.button_filled,
+                    pressed: styles.button_filled_pressed,
+                    disabled: styles.button_filled_disabled,
+                    label: styles.label,
+                },
+                [ButtonType.OUTLINE]: {
+                    style: styles.button_outline,
+                    pressed: styles.button_outline_pressed,
+                    disabled: styles.button_outline_disabled,
+                    label: styles.content_pressed,
+                },
+                [ButtonType.GHOST]: {
+                    style: styles.button_ghost,
+                    pressed: styles.button_ghost_pressed,
+                    disabled: styles.content_disabled,
+                    label: styles.content_pressed,
+                },
+            };
+        }, []);
 
-const Button: React.FC<Properties> = ({ label, onPress }) => {
+    const isFilledButton = buttonType === ButtonType.FILLED;
+    const isGhostButton = buttonType === ButtonType.GHOST;
     return (
-        <Pressable style={styles.btn} onPress={onPress}>
-            <Text style={styles.label}>{label}</Text>
+        <Pressable
+            disabled={disabled}
+            style={({ pressed }): StyleRecord[] => [
+                isGhostButton
+                    ? (globalStyles.borderRadius24,
+                      globalStyles.flexDirectionRow)
```

Comment:
I guess we can add Icon not only to ghost type

Reply:
fix